crates/services/p2p/src/p2p_service.rs

```diff
@@ -1594,6 +1627,46 @@ mod tests {
                                             }
                                         });
                                     }
+                                    RequestMessage::TxPoolAllTransactionsIds => {
```

Comment:
imo, this is a lot of nesting within the same test, can we abstract some commonly used funcionality out?